src/test/java/org/fcrepo/client/MockHttpExpectations.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.client;
+
+import org.apache.http.HttpStatus;
+import org.mockserver.client.server.MockServerClient;
+
+import java.net.URI;
+
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+/**
+ * Expectations for the Mock Http Server
+ *
+ * @author esm
+ */
+public class MockHttpExpectations {
+
+    /**
+     * The TCP host the mock HTTP server listens to.
+     */
+    static String host = "localhost";
+
+    /**
+     * The TCP port the mock HTTP server listens to.
+     */
+    static String port;
+
+    static class Uris {
+        int statusCode;
+        String path;
+
+        Uris(final int statusCode) {
+            this.statusCode = statusCode;
+            this.path = "/uri/" + statusCode;
+        }
+
+        URI asUri() {
+            return URI.create("http://" + host + ":" + port + path);
+        }
+
+        @Override
+        public String toString() {
+            return asUri().toString();
+        }
+    }
+
+    static class SupportedUris {
```

Comment:
The issue is there is state in the Uris class, especially port number, that is only assigned after the `MockHttpExpectations().initializeExpectations(...)` is called.